### PR TITLE
[candi] Use /opt/deckhouse/tmp instead of /tmp during bootstrap

### PR DESCRIPTION
**Pull request by mvasl**
## Description

Temporary files created by bashible steps during bootstrap will now be put into `/opt/deckhouse/tmp` instead of `/tmp`. `dhctl` will create this directory during master node bootstrap.

## Why do we need it, and what problem does it solve?

This solves points 1,3,4 and 5 of [#5597](https://github.com/deckhouse/deckhouse/issues/5597#issuecomment-1695348015)

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi
type: chore
summary: Point $TMPDIR to `/opt/deckhouse/tmp` during bashible bootstrap.
impact_level: default
```

```changes
section: dhctl
type: chore
summary: Render bootstrap bundle into `/opt/deckhouse/tmp`.
impact_level: default
```

```changes
section: candi
type: fix
summary: Delete registrypackage files if it failed to install before retrying installation.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/openstack'...

File: dhctl/pkg/operations/bootstrap.go [hunk 1]
```diff
@@ -55,6 +55,19 @@ func BootstrapMaster(sshClient *ssh.Client, bundleName, nodeIP string, metaConfi
 			return fmt.Errorf("prepare bootstrap: %v", err)
 		}
 
+		err := log.Process("bootstrap", "Create /opt/deckhouse{bin,tmp}", func() error {
+			if err := sshClient.Command("mkdir", "-p", "/opt/deckhouse/bin").Sudo().Run(); err != nil {
```
Comment (RomanenkoDenys) [inline dhctl/pkg/operations/bootstrap.go:59]: Move "/opt/deckhouse/bin" to constant and use it in code

File: dhctl/pkg/operations/bootstrap.go [hunk 2]
```diff
@@ -55,6 +55,19 @@ func BootstrapMaster(sshClient *ssh.Client, bundleName, nodeIP string, metaConfi
...
+				return fmt.Errorf("ssh: mkdir -p /opt/deckhouse/bin: %w", err)
+			}
+			if err := sshClient.Command("mkdir", "-p", "-m", "1777", "/opt/deckhouse/tmp").Sudo().Run(); err != nil {
+				return fmt.Errorf("ssh: mkdir -p -m 1777 /opt/deckhouse/tmp: %w", err)
```
Comment (RomanenkoDenys) [inline dhctl/pkg/operations/bootstrap.go:63]: Move "/opt/deckhouse/tmp" to constant and use it in code

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some minor changes

File: dhctl/pkg/operations/bootstrap.go [hunk 3]
```diff
@@ -55,6 +55,19 @@ func BootstrapMaster(sshClient *ssh.Client, bundleName, nodeIP string, metaConfi
...
+			if err := sshClient.Command("mkdir", "-p", app.DeckhouseNodeBinPath).Sudo().Run(); err != nil {
+				return fmt.Errorf("ssh: mkdir -p %s: %w", app.DeckhouseNodeBinPath, err)
+			}
+			if err := sshClient.Command("mkdir", "-p", "-m", "1777", app.DeckhouseNodeTmpPath).Sudo().Run(); err != nil {
+				return fmt.Errorf("ssh: mkdir -p -m 1777 %s: %w", app.DeckhouseNodeTmpPath, err)
+			}
+			return nil
+		})
+		if err != nil {
+			return fmt.Errorf("cannot create /opt/deckhouse/ directories: %w", err)
```
Comment (RomanenkoDenys) [inline dhctl/pkg/operations/bootstrap.go:68]:
```suggestion
			return fmt.Errorf("cannot create %s directories: %w",NodeDeckhouseDirectoryPath, err)
```

File: dhctl/pkg/system/ssh/frontend/upload-script.go [hunk 4]
```diff
@@ -78,7 +78,7 @@ func (u *UploadScript) Execute() (stdout []byte, err error) {
 
 	remotePath := "."
 	if u.sudo {
-		remotePath = "/tmp/" + scriptName
+		remotePath = "/opt/deckhouse/tmp/" + scriptName
```
Comment (RomanenkoDenys) [inline dhctl/pkg/system/ssh/frontend/upload-script.go:81]: const?

File: dhctl/pkg/system/ssh/frontend/upload-script.go [hunk 5]
```diff
@@ -142,15 +142,15 @@ func (u *UploadScript) ExecuteBundle(parentDir, bundleDir string) (stdout []byte
 
 	tomb.RegisterOnShutdown("Delete bashible bundle folder", func() { _ = os.Remove(bundleLocalFilepath) })
 
-	// upload to /tmp
-	err = NewFile(u.Session).Upload(bundleLocalFilepath, "/tmp")
+	// upload to /opt/deckhouse/tmp
+	err = NewFile(u.Session).Upload(bundleLocalFilepath, "/opt/deckhouse/tmp")
```
Comment (RomanenkoDenys) [inline dhctl/pkg/system/ssh/frontend/upload-script.go:146]: const?

File: dhctl/pkg/system/ssh/frontend/upload-script.go [hunk 6]
```diff
@@ -142,15 +142,15 @@ func (u *UploadScript) ExecuteBundle(parentDir, bundleDir string) (stdout []byte
...
 	if err != nil {
 		return nil, fmt.Errorf("upload: %v", err)
 	}
 
 	// sudo:
-	// tar xpof /tmp/bundle.tar -C /var/lib && /var/lib/bashible/bashible.sh args...
-	tarCmdline := fmt.Sprintf("tar xpof /tmp/%s -C /var/lib && /var/lib/%s/%s %s", bundleName, bundleDir, u.ScriptPath, strings.Join(u.Args, " "))
+	// tar xpof /opt/deckhouse/tmp/bundle.tar -C /var/lib && /var/lib/bashible/bashible.sh args...
+	tarCmdline := fmt.Sprintf("tar xpof /opt/deckhouse/tmp/%s -C /var/lib && /var/lib/%s/%s %s", bundleName, bundleDir, u.ScriptPath, strings.Join(u.Args, " "))
```
Comment (RomanenkoDenys) [inline dhctl/pkg/system/ssh/frontend/upload-script.go:153]: const?

**Review by RomanenkoDenys — COMMENTED** [review]
consts

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Review by name212 — APPROVED** [review]

**Pull request merged**